python/sparkdl/image/imageIO.py

```diff
@@ -118,89 +88,77 @@ def imageStructToArray(imageRow):
-        return SparkMode.FLOAT32
-    elif num_channels != 3:
-        raise ValueError("number of channels of the input array (%d) is not supported" %
-                         num_channels)
-    elif arr.dtype == np.uint8:
-        return SparkMode.RGB
-    elif arr.dtype in [np.float16, np.float32, np.float64]:
-        return SparkMode.RGB_FLOAT32
+    if arr.dtype == "uint8":
+        name = "CV_8UC%d" % num_channels
+    elif arr.dtype == "float32":
+        name = "CV_32FC%d" % num_channels
     else:
-        raise ValueError("did not find a sparkMode for the given array with num_channels = %d " +
-                         "and dtype %s" % (num_channels, arr.dtype.string))
-
-
-def _resizeFunction(size):
-    """ Creates a resize function.
-    
-    :param size: tuple, size of new image: (height, width). 
-    :return: function: image => image, a function that converts an input image to an image with 
-    of `size`.
-    """
-
-    if len(size) != 2:
-        raise ValueError("New image size should have for [hight, width] but got {}".format(size))
-
-    def resizeImageAsRow(imgAsRow):
-        imgAsArray = imageStructToArray(imgAsRow)
-        imgType = imageType(imgAsRow)
-        imgAsPil = Image.fromarray(imgAsArray, imgType.pilMode)
-        imgAsPil = imgAsPil.resize(size[::-1])
-        imgAsArray = np.array(imgAsPil)
-        return imageArrayToStruct(imgAsArray, imgType.sparkMode)
-
-    return resizeImageAsRow
+        raise ValueError("Unsupported type '%s'" % arr.dtype)
+    return imageTypeByName(name)
 
-
-def resizeImage(size):
+def resizeImage_python(size):
     """ Create a udf for resizing image.
-    
+
     Example usage:
     dataFrame.select(resizeImage((height, width))('imageColumn'))
-    
-    :param size: tuple, target size of new image in the form (height, width). 
+
+    :param size: tuple, target size of new image in the form (height, width).
     :return: udf, a udf for resizing an image column to `size`.
     """
-    return udf(_resizeFunction(size), imageSchema)
+    if len(size) != 2:
+        raise ValueError("New image size should have for [height, width] but got {}".format(size))
+    def _resizeImageAsRow(imgAsRow):
+        if (imgAsRow.height,imgAsRow.width) == size:
+            return imgAsRow
+        imgAsPil = imageStructToPIL(imgAsRow).resize(size[::-1])
+        imgAsArray = (np.array(imgAsPil)[...,::-1])
+        return imageArrayToStruct(imgAsArray,origin=imgAsRow.origin)
+    return udf(_resizeImageAsRow, ImageSchema.imageSchema['image'].dataType)
 
 
-def _decodeImage(imageData):
-    """
-    Decode compressed image data into a DataFrame image row.
+# cache for created udfs
+__resize_udfs = {}
 
-    :param imageData: (bytes, bytearray) compressed image data in PIL compatible format.
-    :return: Row, decoded image.
-    """
-    try:
-        img = Image.open(BytesIO(imageData))
-    except IOError:
-        return None
+def resizeImage_jvm(size):
```

Comment:
Discussed offline: Let's go ahead and remove this method, since it's not currently being used.